src/Clauses/OptionalMatchClause.php

```diff
@@ -30,19 +32,23 @@
  */
 class OptionalMatchClause extends Clause
 {
+    use ErrorTrait;
+
     /**
-     * @var StructuralType[] List of patterns
+     * @var StructuralType[]|Assignment[] List of patterns
      */
     private array $patterns = [];
 
     /**
      * Add a pattern to the optional match clause.
      *
-     * @param StructuralType $pattern
+     * @param StructuralType|Assignment $pattern
      * @return OptionalMatchClause
      */
-    public function addPattern(StructuralType $pattern): self
+    public function addPattern( $pattern ): self
```

Comment:
Please format this using the PSR-2 coding standard (see https://www.php-fig.org/psr/psr-2/#43-methods)